src/main/kotlin/io/bazel/kotlin/builder/tasks/jvm/CompilationTask.kt

```diff
@@ -165,6 +165,53 @@ internal fun JvmCompilationTask.kaptArgs(
   }
 }
 
+internal fun JvmCompilationTask.kspArgs(
+  plugins: InternalCompilerPlugins,
+): CompilationArgs {
+  return CompilationArgs().apply {
+    plugin(plugins.kspSymbolProcessingCommandLine)
+    plugin(plugins.kspSymbolProcessingApi) {
+      flag("-Xallow-no-source-files")
+
+      val values = arrayOf(
+        "apclasspath" to listOf(inputs.processorpathsList.joinToString(File.pathSeparator)),
+        // projectBaseDir shouldn't matter because incremental is disabled
+        "projectBaseDir" to listOf(directories.incrementalData),
+        // Disable incremental mode
+        "incremental" to listOf("false"),
+        // Directory where class files are written to. Files written to this directory are class
+        // files being written directly from the annotation processor, not Kotlinc
+        "classOutputDir" to listOf(directories.generatedClasses),
```

Comment:
`classOutputDir`, `javaOutputDir`, and `javaOutputDir` are currently being ignored when written to. We'll need to look into adding support for this incrementally.